src/filters/sse.rs

```diff
@@ -449,21 +451,36 @@ where
 }
 
 #[allow(missing_debug_implementations)]
-struct SseKeepAlive<S> {
+/// Struct for tracking the interval between keep-alive messages, the content
+/// of each message, and the associated stream.
+pub struct SseKeepAlive<S> {
     event_stream: S,
+    comment_text: String,
     max_interval: Duration,
     alive_timer: Delay,
 }
+impl<S> SseKeepAlive<S> {
+    /// Customize the interval between keep=alive messages.
+    pub fn with_interval(mut self, time: Duration) -> Self {
+        self.max_interval = time;
+        self
+    }
 
-/// Keeps event source connection when no events sent over a some time.
-///
-/// Some proxy servers may drop HTTP connection after a some timeout of inactivity.
-/// This function helps to prevent such behavior by sending dummy events with single
-/// empty comment field (i.e. ":" only) each `keep_interval` of inactivity.
-///
-/// By default this time interval between events equals to 15 seconds.
+    /// Customize the text of the keep-alive message.
+    pub fn with_text(mut self, text: String) -> Self {
```

Comment:
I don't have a strong opinion here, but: it looks like `sse::comment` is able to be any `Display` type. Should that be allowed here also?